### PR TITLE
improve service support, + miscellaneous changes

### PR DESCRIPTION
**Pull request by mazunki**
- write pidfile when running daemon
- add logger info to pidfile
- catch SIGHUP signal to reload config during runtime
- use dist/ for packaging purposes
- add openrc service
- autoconf ignore
- organize files, support autoconf
- change some flags to subcommands
- ignore wheels
- change entry point to main
- update documentation
- update tests. permission denied claims 2==1?

### CODE REVIEW

**Comment by mazunki** [comment]
i think it would be wise to update and aggregate to the tests too. i only updated the tests which already existed

**Comment by mcgillij** [comment]
Cool I'll take a look, sorry was out on vacation this week, so haven't kept up with GH.

File: INSTALL.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+# Configuration files
+Similiar to above, you can obtain the sources from the runtime. This is not necessary, as this file is generated automatically after the daemon runs for the first time.
+
+``` bash
+amdfan print-default --configuration | sudo tee /etc/amdfan.yml
+```
+
+
+# Executable files
+The executable files are contained within amdfan. This directory is a python module, and can either be loaded as `python -m amdgpu`. If you want to import the module, you may be interested in checking out `amdfan.commands`, which contains most of the subcommands.
```
Comment (mcgillij) [inline INSTALL.md:38]:
```suggestion
The executable files are contained within amdfan. This directory is a python module, and can either be loaded as `python -m amdfan`. If you want to import the module, you may be interested in checking out `amdfan.commands`, which contains most of the subcommands.
```

**Comment by mcgillij** [comment]
Looks good, I'll have to pull it down to make sure I can still build it for the Arch packaging, and to make sure it still works with my [py3status-amdfan](https://github.com/mcgillij/py3status-amdfan/)

**Comment by mcgillij** [comment]
yeah I fixed at least the existing tests, so that the right cli funcs are passed so we no longer have the 2 == 1 etc.

The rest of the changes were just to make it work with the github workflow build.

**Pull request merged**